innovation.game.php

```diff
@@ -10651,6 +10669,32 @@ function stInteractionStep() {
             );
             break;
 
+        // id 129, Artifacts age 2: Holy Lance
+        case "129C1A":
+            // Find colors with top artifacts
+            $artf_colors = array();
+            for ($color = 0; $color < 5; $color++)
+            {
+                $top_card = self::getTopCardOnBoard($player_id, $color);
+                if ($top_card !== null && $top_card['type'] == 1) {
+                    $artf_colors[] = $color;
+                }
+            }
```

Comment:
Instead of using a separate query to fetch the top Artifacts and needing to replicate this code when we want to do this in other places, I've added an option like `'type' => array(1)` that we can use.